src/components/LinkChooser/LinkChooser.tsx

```diff
@@ -0,0 +1,218 @@
+        if (foundItem) send("SET_SELECTED_SEARCH_RESULT", { data: { selectedResult: foundItem } });
+    };
+
+    const handleInputChange = (value: string) => {
+        send("TYPING", { data: { query: value } });
+    };
+
+    const handleDropdownOpen = () => {
+        send("OPEN_DROPDOWN");
+    };
+
+    const handleDropdownClose = () => {
+        const selectedResult = context.query
+            ? queryMatchesSelection(context.selectedResult, context.query)
+                ? context.selectedResult
+                : createCustomLink(state.inputValue)
+            : null;
+        send("CLOSE_DROPDOWN", { data: { selectedResult } });
+        state.setSelectedKey(selectedResult?.id || "");
+    };
+
+    const filterItems = (value: string, query: string) => value.toLowerCase().includes(query.toLowerCase());
+
+    const searchResultMenuBlock = useMemo(
+        () => [
+            {
+                id: "search",
+                menuItems: context.searchResults,
+            },
+        ],
+        [context.searchResults],
+    );
+
+    const props = mapToAriaProps(ariaLabel, searchResultMenuBlock);
+
+    const state = useComboBoxState({
+        ...props,
+        defaultFilter: filterItems,
+        onInputChange: handleInputChange,
+        onSelectionChange: handleSelectionChange,
+        menuTrigger: "manual",
+        allowsEmptyCollection: true,
+        onBlur: handleDropdownClose,
+    });
+
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const listBoxRef = useRef<HTMLUListElement | null>(null);
+    const popoverRef = useRef<HTMLDivElement | null>(null);
+
+    const { inputProps, listBoxProps, labelProps } = useComboBox(
+        {
+            ...props,
+            inputRef,
+            listBoxRef,
+            popoverRef,
+            onBlur: handleDropdownClose,
+        },
+        state,
+    );
+
+    const formattedIcon = context.selectedResult?.icon
+        ? ICON_OPTIONS[context.selectedResult.icon]
+        : ICON_OPTIONS[DEFAULT_ICON];
+
+    useEffect(() => {
+        if (isLoaded(matches) && context.interruptedFetch) {
+            send("TYPING", { data: { query: context.query } });
+        }
+    }, [context.interruptedFetch, value]);
+
+    return (
+        <div data-test-id="link-chooser" className="tw-relative tw-w-full tw-font-sans tw-text-s">
+            <label {...labelProps} className="block text-sm font-medium text-gray-700 text-left">
+                {label}
+            </label>
+            <div
+                className={`relative inline-flex flex-row rounded-md overflow-hidden shadow-sm border-2 ${
+                    state.isFocused ? "border-pink-500" : "border-gray-300"
+                }`}
+            >
+                <SearchInput
+                    {...inputProps}
+                    selectedResult={context.selectedResult}
+                    ref={inputRef}
+                    decorator={formattedIcon}
+                    previewable={true}
```

Comment:
are these props always true? imho this sounds like a check which can be done by the state

Reply:
removed and set them to `true` by default 👍